src/shell.rs

```diff
@@ -148,7 +153,15 @@ impl MessageInfo {
             verbosity,
             stdout_needs_erase: false,
             stderr_needs_erase: false,
+            cross_debug: false,
+        }
+    }
+
+    pub fn cross_debug_check(mut self) -> Self {
```

Comment:
I don't particularly like this name. It should probably be `cross_debug` and then ensure it's always called rather than the field used (with a `&mut self` signature), put into the constructor, or something similar. This I don't feel properly describes what it does.